accounts/tests.py

```diff
@@ -127,23 +156,156 @@ def test_account_rejection(self, mock_post, mock_get):
-        requests = AccountRequests.objects.all()
+        requests = AccountRequests.objects.filter(agol_id=None)
         invitations = agol.generate_invitations(requests, 'password')
-        self.assertTrue('000b6ee121bb4739a5021e0f0241ff01' in invitations[0]["groups"])
-        self.assertTrue('00237b674c3347a18e0fb2bfcdb248e1' in invitations[0]["groups"])
+        self.assertTrue(requests[0].email in invitations[0]["email"])
+        self.assertTrue(requests[0].first_name in invitations[0]["firstname"])
+        self.assertTrue(requests[0].last_name in invitations[0]["lastname"])
+
+    def test_user_has_account_request_permission(self):
+        request = self.factory.get('/account/approvals')
+
+        # does sponsor have permission
+        sponsor = User.objects.get(pk=1)
+
+        request.user = sponsor
+        for ac in AccountRequests.objects.all():
+            result = IsSponsor().has_object_permission(request, SponsorsViewSet, ac)
+            self.assertTrue(result)
+
+        # does unrelated user have permission
+        unrelated_user = User.objects.get(pk=2)
+        request.user = unrelated_user
+        for ac in AccountRequests.objects.all():
+            result = IsSponsor().has_object_permission(request, SponsorsViewSet, ac)
+            self.assertFalse(result)
+
+        # after making delegate does user have permission
+        sponsor.agol_info.delegates.set([unrelated_user])
+        for ac in AccountRequests.objects.all():
+            result = IsSponsor().has_object_permission(request, SponsorsViewSet, ac)
+            self.assertTrue(result)
+
+
+class TestAGOLGroups(TestCase):
+    fixtures = ['fixtures.json']
+
+    # Get AGOLGroups by param tests #
+    @patch.dict('os.environ', {'all': 'true'})
+    def test_get_agol_groups_by_all_param(self):
+        if 'all' in os.environ:
+            if os.environ.get('all', 'false') == 'true':
+                all_results = list(AGOLGroup.objects.all())
+                self.assertIsNotNone(all_results)
+
+    @patch.dict('os.environ', {'is_auth_group': 'true'})
+    def test_get_agol_groups_by_is_auth_group_param(self):
+        if 'is_auth_group' in os.environ:
+            is_auth_group = os.environ['is_auth_group']
+            if is_auth_group == 'true':
+                is_auth_group = True
+            elif is_auth_group == 'false':
+                is_auth_group = False
+            is_auth_group_results = list(AGOLGroup.objects.filter(is_auth_group=is_auth_group))
+            self.assertIsNotNone(is_auth_group_results)
+
+    def test_get_agol_groups_by_sponsor(self):
+        sponsor = User.objects.get(pk=1)
+        sponsors = User.objects.filter(agol_info__delegates=sponsor)
+        sponsor_results = AGOLGroup.objects.filter(Q(response__users=sponsor) | Q(response__users__in=sponsors))
+        self.assertTrue(len(sponsor_results) > 0)
+
+
+class TestSponsor(TestCase):
+    fixtures = ['fixtures.json']
 
+    # Get Sponsors by param tests #
```

Comment:
@tommckennon what are you testing here exactly? This really looks like its just testing django/drf code.  If you want to just make sure a certain field is set in search_fields for that viewset... do that directly... do not testing django/drf

Reply:
Removed